[ews] Expected additionalArguments frequently missing from iOS layout test queues

aj062 opened the pull request:
#### d573b83aa2c218b96c855021965c9a683f231274
<pre>
[ews] Expected additionalArguments frequently missing from iOS layout test queues
<a href="https://bugs.webkit.org/show_bug.cgi?id=255742">https://bugs.webkit.org/show_bug.cgi?id=255742</a>

Reviewed by Ryan Haddad.

additionalArguments were missing because in CompileWebKit class (and sub-classes), we were inadvertently removing
them. Especially compile-webkit-without-change step was the one removing the additionalArguments. This accessed build property
and removed various values from it, buy since a list is accessed by reference, original property was also modified.

* Tools/CISupport/ews-build/steps.py:
(CompileWebKit): Do not explicitly remove arguments, instead just check if the additionalArguments is within valid list for that step.
(RunWebKitTests): Drive-by fix. Code refactoring to adopt clener approach to handle additionalArguments, using ENABLE_ADDITIONAL_ARGUMENTS.
(RunWebKitTestsInStressMode): Ditto.
(RunWebKitTestsInStressMode.setLayoutTestCommand): Ditto.
(Trigger.propertiesToPassToTriggers): Revert previous attempt to fix this issue.

Canonical link: <a href="https://commits.webkit.org/263387@main">https://commits.webkit.org/263387@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/72701f6b0280249837a0c593765f01da4c06ca44

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4433 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4553 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4690 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5912 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/4630 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4422 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4679 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4517 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/4860 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4497 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/4634 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/3983 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/5914 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/2129 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/3962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/8178 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/3971 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/4031 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/5574 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/4439 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/3616 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/4395 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/3964 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/3961 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/8014 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/510 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4318 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -2276,7 +2276,6 @@ def propertiesToPassToTriggers(self, patch=True, pull_request=False):
             'platform',
             'fullPlatform',
             'architecture',
-            'additionalArguments',
```
This is simply reverting previous attempt to fix this issue (https://commits.webkit.org/263195@main).

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3308,6 +3304,7 @@ class RunWebKitTests(shell.Test, AddToLogMixin):
     test_failures_log_name = 'test-failures'
     results_db_log_name = 'results-db'
     ENABLE_GUARD_MALLOC = False
+    ENABLE_ADDITIONAL_ARGUMENTS = True
```
This (and other changes in RunWebKitTests and RunWebKitTestsInStressMode) are not strictly needed to fix the issue. But these are code refactoring to adopt cleaner approach to handle additionalArguments in RunWebKitTestsInStressMode.

ryanhaddad reviewed (approved):
Wow, I didn't realize that the CompileWebKit step could do this when it wasn't explicitly run. (edit: Aakash explained on Slack that this pathology appeared as soon as this step was run once on the queue, it removed the arguments for all subsequent builds)

Thanks for figuring this out, Aakash.

aj062 committed:
[ews] Expected additionalArguments frequently missing from iOS layout test queues
https://bugs.webkit.org/show_bug.cgi?id=255742

Reviewed by Ryan Haddad.

additionalArguments were missing because in CompileWebKit class (and sub-classes), we were inadvertently removing
them. Especially compile-webkit-without-change step was the one removing the additionalArguments. This accessed build property
and removed various values from it, buy since a list is accessed by reference, original property was also modified.

* Tools/CISupport/ews-build/steps.py:
(CompileWebKit): Do not explicitly remove arguments, instead just check if the additionalArguments is within valid list for that step.
(RunWebKitTests): Drive-by fix. Code refactoring to adopt clener approach to handle additionalArguments, using ENABLE_ADDITIONAL_ARGUMENTS.
(RunWebKitTestsInStressMode): Ditto.
(RunWebKitTestsInStressMode.setLayoutTestCommand): Ditto.
(Trigger.propertiesToPassToTriggers): Revert previous attempt to fix this issue.

Canonical link: https://commits.webkit.org/263387@main
